src/mpbuild/build.py

```diff
@@ -2,37 +2,30 @@
 
 import subprocess
 
-# ['esp8266', 'esp32', 'samd', 'nrf', 'renesas-ra', 'mimxrt', 'stm32', 'cc3200', 'rp2']
-
-_arm_build_container = "micropython/build-micropython-arm"
-_build_containers = {
-    "stm32": _arm_build_container,
-    "rp2": _arm_build_container,
-    "nrf": _arm_build_container,
-    "mimxrt": _arm_build_container,
-    "renesas-ra": _arm_build_container,
-    "samd": _arm_build_container,
+ARM_BUILD_CONTAINER = "micropython/build-micropython-arm"
+BUILD_CONTAINERS = {
+    "stm32": ARM_BUILD_CONTAINER,
+    "rp2": ARM_BUILD_CONTAINER,
+    "nrf": ARM_BUILD_CONTAINER,
+    "mimxrt": ARM_BUILD_CONTAINER,
+    "renesas-ra": ARM_BUILD_CONTAINER,
+    "samd": ARM_BUILD_CONTAINER,
     "esp32": "espressif/idf",
     "unix": "micropython/build-micropython-unix",  # Special, doesn't have boards
 }
 
 IDF_DEFAULT = "v5.2.2"
 
-# docker run -it --rm -e HOME=/tmp  -e BOARD=$BOARD -e ARGS="$ARGS" -e
-# DEPLOY_PORT=$DEPLOY_PORT -e UID=$(id -u) -v /sys/bus:/sys/bus -v /dev:/dev
-# --net=host --privileged -v "$CD":"$CD" -w "$CD" --user 1000:1000
-# espressif/idf:$IDF_VER bash -c "git config --global --add safe.directory
-# '*';make -C ports/esp32 BOARD=ESP32_GENERIC_C3"
-
 
 def build_board(
     port: str,
     board: str,
     variant: Optional[str] = None,
     extra_args: Optional[List[str]] = [],
+    build_container_override: Optional[str] = None,
     idf: Optional[str] = None,
 ) -> None:
-    if port not in _build_containers.keys():
+    if port not in BUILD_CONTAINERS.keys():
         print(f"Sorry, builds are not supported for the {port} port at this time")
         # TODO(mst) Should raise an exception and abort with an error code
         return
```

Comment:
FYI to raise exception to abort with an error code you can simply use:
``` python
raise SystemExit()
```

Reply:
Updated